timer.js

```diff
@@ -42,5 +49,124 @@ function stop() {
 function reset() {
     
 }
-// to test that timer works
-window.onload = function() { startTimer(pomdoro_duration) };
\ No newline at end of file
+function setTime() {
+    if (modeDisplay.textContent == 'Work Session') {
+        minutes = workMinutesSetting.textContent;
+        duration = minutes*60;
+    }
+    else {
+        minutes = breakMinutesSetting.textContent;
+        duration = minutes*60;
+    }
+    timer = duration, minutes, seconds;
+}
+
+
+var intervalCount;
+var workMinutesSetting = document.getElementById('work-minutes');
+var breakMinutesSetting = document.getElementById('break-minutes');
+var modeDisplay = document.getElementById('mode');
+var minutesDisplay = document.getElementById('minutes');
+var secondsDisplay = document.getElementById('seconds');
+var timer; 
+var minutes; 
+var seconds = secondsDisplay.textContent;;
+var intervalId;
+var intervalCount = -1;
+
+// add event listeners for play button
+var playButton = document.getElementById('play');
+if (playButton.addEventListener) {
```

Comment:
what does the conditional check? That the button wants a listener? 😮 🤔

Reply:
A conditional check that you can use that method for the browser:

"You can also add event listeners using a method called addEventListener. However, this method isn't supported in Internet Explorer 8, so if you use this method, you need to add some conditional functions to check for browser functionality before running the function."

https://idratherbewriting.com/events-and-listeners-javascript/

:)

Reply:
Ooh thanks, B!